odoo/models.py

```diff
@@ -5521,6 +5521,33 @@ def onchange(self, values, field_name, field_onchange):
 
         return result
 
+    #
+    # Cleanup
+    #
+
+    @api.model
+    def _expunge_bad_records(self):
+        """
+        Expunge bad records from the current model.
+
+        This method, when overridden, is used to remove records of models
+        which may have database constraints and could not be removed normally
+        through the unlink method.
```

Comment:
Mmm the description looks very specific, while it is actually quite generic. What about this?

    This method, when overridden, may be used to clean up or remove records that
    were related to other data that is gone because of some module deinstallation.
    It typically fixes failed removals because of database constraints or the absence
    of such constraints.